jetstream_pt/engine.py

```diff
@@ -447,6 +456,45 @@ def insert(
         slot,
     )
 
+  def precompute_ragged_block_indices(self, decode_state: DecodeState):
```

Comment:
Nit: do you need other attribute from decode_state besides start and input_pos, pass start and input_pos instead of heavy object decode_state.

Reply:
Only the start and input_pos is used, but considering we just passing the reference to the decode_state, so it should not affect the performance at all.

Reply:
sounds good